types/src/crypto/asymmetric_key.rs

```diff
@@ -443,6 +443,21 @@ impl SecretKey {
     }
 }
 
+#[cfg(any(feature = "testing", test))]
+impl rand::distributions::Distribution<SecretKey> for rand::distributions::Standard {
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> SecretKey {
+        if rng.gen() {
+            let mut bytes = [0u8; SecretKey::ED25519_LENGTH];
+            rng.fill_bytes(&mut bytes[..]);
+            SecretKey::ed25519_from_bytes(bytes).unwrap()
+        } else {
+            let mut bytes = [0u8; SecretKey::SECP256K1_LENGTH];
+            rng.fill_bytes(&mut bytes[..]);
+            SecretKey::secp256k1_from_bytes(bytes).unwrap()
+        }
+    }
+}
+
```

Comment:
I'd prefer if we didn't add this.  We've had to add some impls like this to support e.g. proptest tests, but I prefer to stick to the convention of providing a `random` function which takes a `TestRng` specifically, to try to avoid usage of other rngs which don't dump a seed on failure, making test failures harder to reproduce.